zerver/views/realm_aliases.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+
+from django.core.exceptions import ValidationError
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+
+from zerver.decorator import has_request_variables, REQ, require_realm_admin
+from zerver.lib.response import json_error, json_success
+from zerver.lib.validator import check_string
+from zerver.models import can_add_alias, RealmAlias, UserProfile
+
+from typing import Text
+
+def list_aliases(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> (HttpResponse)
+    # return aliases
+    aliases = []
+    for alias in RealmAlias.objects.filter(realm=user_profile.realm):
+        aliases.append((alias.id, alias.domain))
+    return json_success({'aliases': aliases})
+
+@require_realm_admin
+@has_request_variables
+def create_alias(request, user_profile, domain=REQ()):
+    # type: (HttpRequest, UserProfile, Text) -> (HttpResponse)
+    if can_add_alias(domain):
+        try:
+            alias = RealmAlias(realm=user_profile.realm, domain=domain)
+            alias.full_clean()
+            alias.save()
+        except ValidationError:
+            return json_error(_('Domain can\'t be empty.'))
+    else:
+        return json_error(_('A Realm for this domain already exists.'))
+    return json_success({'new_alias': [alias.id, alias.domain]})
+
+@require_realm_admin
+@has_request_variables
+def delete_alias(request, user_profile, alias_id):
+    # type: (HttpRequest, UserProfile, int) -> (HttpResponse)
+    try:
+        RealmAlias.objects.get(pk=alias_id).delete()
+    except RealmAlias.DoesNotExist:
+        return json_error(_('No such entry found.'))
+    return json_success()
```

Comment:
These view functions should use the standard Zulip real-time sync approach of calling an `actions.py` function sending an event to update open browser windows.  

Read http://zulip.readthedocs.io/en/latest/new-feature-tutorial.html and the other tutorials to learn more about how this works.  Or follow how RealmEmoji and RealmFilters work.